src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -157,7 +163,299 @@ public SPDX30Parser(
         return null;
     }
 
-    public Spdx22Metadata GetMetadata()
+    private void ValidateContext(ContextsResult result)
+    {
+        if (result.Contexts == null || !result.Contexts.Any() || result.Contexts.Count() > 1)
+        {
+            throw new ParserException($"The context property is either empty or has more than one string.");
+        }
+    }
+
+    /// <summary>
+    /// Converts JSON objects to SPDX elements.
+    /// </summary>
+    /// <param name="jsonList"></param>
+    /// <returns></returns>
+    public ElementsResult ConvertToElements(List<object>? jsonList, ref ParserStateResult? result, string? requiredComplianceStandard, IReadOnlyCollection<string>? entitiesWithDifferentNTIARequirements)
+    {
+        var elementsResult = new ElementsResult(result);
+        var elementsList = new List<Element>();
+        var elementsSpdxIdList = new HashSet<string>();
+        var filesList = new List<Parsers.Spdx30SbomParser.Entities.File>();
+
+        if (jsonList is null)
+        {
+            elementsResult.FilesCount = 0;
+            elementsResult.PackagesCount = 0;
+            elementsResult.ReferencesCount = 0;
+            elementsResult.RelationshipsCount = 0;
+            elementsResult.Elements = elementsList;
+            return elementsResult;
+        }
+
+        var complianceStandardAsEnum = ComplianceStandard.None;
+        if (!string.IsNullOrEmpty(this.RequiredComplianceStandard))
```

Comment:
Lines 197-218 look a lot like lines 77-98. Can we put some of this into a helper method?